### PR TITLE
Skip deploy when configuration is null

### PR DESCRIPTION
**Pull request by Flanker32**
Skip deploy when configuration is null

### CODE REVIEW

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ protected void initializeAuthConfiguration() throws MojoFailureException {
     }
 
     protected boolean isAuthConfigurationExist() {
-        final String pluginKey = String.format("%s:%s", plugin.getGroupId(), plugin.getArtifactId());
+        final String pluginKey = getPluginKey();
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: plugin.getPluginLookupKey?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 2]
```diff
@@ -227,6 +230,17 @@ protected boolean checkProjectPackaging(MavenProject project) throws MojoExecuti
         }
     }
 
+    protected boolean checkConfiguration() {
+        final String pluginKey = getPluginKey();
+        final Xpp3Dom pluginDom = MavenUtils.getPluginConfiguration(project, pluginKey);
+        if (pluginDom == null) {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: child count = 0

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: Fixed

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 2 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Added

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java (hunk 1 above)
Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: plugin has the method of getPluginLookupKey

**Pull request merged**